pandas/core/arrays/interval.py

```diff
@@ -629,7 +631,19 @@ def __iter__(self):
     def __len__(self) -> int:
         return len(self._left)
 
-    def __getitem__(self, key):
+    @overload
+    def __getitem__(self, key: int | np.integer) -> Interval:
```

Comment:
can also be np.nan i think

Reply:
Not for `IntervalArray` or any other `ExtensionArray` :
```python
>>> pd.array([pd.Interval(1,2), pd.Interval(3,4)])[np.nan]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\Code\pandas_dev\pandastypes\pandas\core\arrays\interval.py", line 648, in __getitem__
    left = self._left[key]
IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

```python
>>> pd.array([1,2,3])[np.nan]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "c:\Code\pandas_dev\pandastypes\pandas\core\arrays\masked.py", line 159, in __getitem__
    return type(self)(self._data[item], self._mask[item])
IndexError: only integers, slices (`:`), ellipsis (`...`), numpy.newaxis (`None`) and integer or boolean arrays are valid indices
```

Reply:
i meant it can return np.nan

Reply:
got it.  Fixed in next commit